modules/custom/chatbot_migration/src/Plugin/migrate/source/Evento.php

```diff
@@ -0,0 +1,158 @@
+
+use Drupal\migrate\Plugin\migrate\source\SqlBase;
+use Drupal\migrate\Row;
+use Drupal\migrate\Plugin\MigrationInterface;
+
+/**
+ * D8 nodes from D7 database.
+ *
+ * @MigrateSource(
+ *   id = "evento"
+ * )
+ */
+class Evento extends SqlBase {
+  public function query() {
+    $query = $this->select('node', 'n')
+      ->fields('n', array(
+        'nid',
+        'vid',
+        'title'
+      ));
+
+    /* Join tables for getting fields.*/
+    // For text, link and email fields.
+    $query->leftJoin('field_data_body', 'fdb', 'fdb.entity_id = n.nid AND fdb.revision_id = n.vid');
+    $query->leftJoin('field_data_field_es_destacado', 'fdfed', 'fdfed.entity_id = n.nid AND fdfed.revision_id = n.vid');
+    $query->leftJoin('field_data_field_precio', 'fdfp', 'fdfp.entity_id = n.nid AND fdfp.revision_id = n.vid');
+    $query->leftJoin('field_data_field_nombre_del_lugar', 'fdfndl', 'fdfndl.entity_id = n.nid AND fdfndl.revision_id = n.vid');
+    $query->leftJoin('field_data_field_direccion_del_lugar', 'fdfddl', 'fdfddl.entity_id = n.nid AND fdfddl.revision_id = n.vid');
+    $query->leftJoin('field_data_field_sitio_web', 'fdfsw', 'fdfsw.entity_id = n.nid AND fdfsw.revision_id = n.vid');
+    $query->leftJoin('field_data_field_telefono_de_contacto', 'fdftdc', 'fdftdc.entity_id = n.nid AND fdftdc.revision_id = n.vid');
+    $query->leftJoin('field_data_field_email_de_contacto', 'fdfedc', 'fdfedc.entity_id = n.nid AND fdfedc.revision_id = n.vid');
+    $query->leftJoin('field_data_field_fecha_y_hora', 'fdffyh', 'fdffyh.entity_id = n.nid AND fdffyh.revision_id = n.vid');
+    // For fields referencing taxonomies.
+    $query->leftJoin('field_data_field_categoria_evento', 'fdfce', 'fdfce.entity_id = n.nid AND fdfce.revision_id = n.vid');
+
+    /* Add fields for each table.*/
+    // field_categoria_evento.
+    $query->addField('fdfce', 'field_categoria_evento_tid');
+    // field_es_destacado.
+    $query->addField('fdfed', 'field_es_destacado_value');
+    // body.
+    $query->addField('fdb', 'body_value');
+    // field_precio.
+    $query->addField('fdfp', 'field_precio_value');
+    // field_nombre_del_lugar.
+    $query->addField('fdfndl', 'field_nombre_del_lugar_value');
+    // field_direccion_del_lugar.
+    $query->addField('fdfddl', 'field_direccion_del_lugar_value');
+    // field_sitio_web.
+    $query->addField('fdfsw', 'field_sitio_web_url');
+    // field_telefono_de_contacto.
+    $query->addField('fdftdc', 'field_telefono_de_contacto_value');
+    // field_email_de_contacto.
+    $query->addField('fdfedc', 'field_email_de_contacto_email');
+    // field_fecha_y_hora.
+    $query->addField('fdffyh', 'field_fecha_y_hora_value');
+    $query->addField('fdffyh', 'field_fecha_y_hora_value2');
+
+    $query->condition('n.type', 'evento');
+    $query->condition('n.status', 1);
+
+    return $query;
+  }
+
+  public function fields() {
+    $fields = [
+      'nid' => t('Node id.'),
```

Comment:
I'm not quite sure; but I think `$this->t()` should work in this context. Could you try it? Please 
If it works; that's the best way to do it